### PR TITLE
Closes #457, adds option to send pre and post acceptance reminder ema…

### LINKED ISSUE #457: Resend Review Emails
- Option to send a manual reminder email
- If a review has not been accepted, add an option to gently remind the user of the request making the option to decline obvious as it’s better the editor know the person will not review the paper than leave the request open.

### PR DESCRIPTION
**Pull request by ajrbyers**
…ils manually

### CODE REVIEW

**Comment by ajrbyers** [comment]
This is not ready for review, I need to add a test for the new decorator first.

File: src/review/views.py [hunk 1]
```diff
@@ -19,9 +19,11 @@
...
 from events import logic as event_logic
 from review import models, logic, forms, hypothesis
-from security.decorators import editor_user_required, reviewer_user_required, reviewer_user_for_assignment_required, \
-    article_edit_user_required, file_user_required, article_decision_not_made, article_author_required, \
-    editor_is_not_author, senior_editor_user_required, section_editor_draft_decisions
+from security.decorators import editor_user_required, reviewer_user_required, \
+    reviewer_user_for_assignment_required,  article_edit_user_required, \
+    file_user_required, article_decision_not_made, article_author_required, \
+    editor_is_not_author, senior_editor_user_required, \
+    section_editor_draft_decisions, article_stage_review_required
```
Comment (mauromsl) [inline src/review/views.py]: I can't see where "article_stage_review_required" lives
```
$ git grep "def article_stage_"
src/security/decorators.py:def article_stage_production_required(func):
src/security/decorators.py:def article_stage_accepted_or_later_required(func):
src/security/decorators.py:def article_stage_accepted_or_later_or_staff_required(func):
```

I suspect this is a new decorator that you forgot commit :P

Comment (ajrbyers) [reply src/review/views.py]: https://github.com/BirkbeckCTP/janeway/pull/831/files

I can see it in the commit...

Comment (mauromsl) [reply src/review/views.py]: Where?

Comment (mauromsl) [reply src/review/views.py]: Ah, GitHub decided to no show me the diff by default

File: src/security/decorators.py [hunk 2]
```diff
@@ -931,6 +1174,27 @@ def wrapper(request, *args, **kwargs):
...
+def article_stage_review_required(func):
+    """
+    Checks that the article is in one of the review stages
+    :param func: func
+    :return: PermissionDenied or func
+    """
+    def review_required_wrapper(request, *args, **kwargs):
+        article = get_object_or_404(
+            models.Article,
+            pk=kwargs['article_id']
```
Comment (mauromsl) [inline src/security/decorators.py]: Could throw a KeyError.

File: src/security/decorators.py [hunk 3]
```diff
@@ -697,13 +866,22 @@ def wrapper(request, *args, **kwargs):
...
             return func(request, *args, **kwargs)
 
-        typeset_assignment = get_object_or_404(production_models.TypesetTask, pk=typeset_id)
-        if request.user == typeset_assignment.typesetter and not typeset_assignment.completed and \
+        typeset_assignment = get_object_or_404(
+            production_models.TypesetTask,
+            pk=typeset_id
+        )
+        if request.user == typeset_assignment.typesetter and\
+                not typeset_assignment.completed and \
```
Comment (mauromsl) [inline src/security/decorators.py]: Backslashes are evil and operator should be at the beginning of the line, is autopep8 doing this?

I think we should leave this kind of task for its own issue instead.

File: src/security/decorators.py (hunk 2 above)
Comment (ajrbyers) [reply src/security/decorators.py]: it would 404 before a key error.

Comment (mauromsl) [reply src/security/decorators.py]: Definately not, `__getitem__` will be invoked on kwargs before `get_object_or_404` is called

Comment (ajrbyers) [reply src/security/decorators.py]: handler and test added

File: src/security/decorators.py (hunk 3 above)
Comment (ajrbyers) [reply src/security/decorators.py]: Dropped autopep8 crap from this module.

**Comment by ajrbyers** [comment]
Requested changes done.

File: src/security/decorators.py [hunk 4]
```diff
@@ -952,3 +953,29 @@ def deny_access(request, *args, **kwargs):
...
+
+def article_stage_review_required(func):
+    """
+    Checks that the article is in one of the review stages
+    :param func: func
+    :return: PermissionDenied or func
+    """
+
+    def review_required_wrapper(request, *args, **kwargs):
+        article_id = kwargs.get('article_id', None)
```
Comment (mauromsl) [inline src/security/decorators.py]: This means that the article_id MUST be a kwarg.
If that is the case, python3 introduced a cool feature to express this in the function signature:

```
def review_required_wrapper(request, *args, article_id=None,**kwargs):
     if article_id is None:
          raise Http404
     [...]
```

Comment (ajrbyers) [reply src/security/decorators.py]: Further changes made.

**Pull request merged**